parcllabs/services/parcllabs_service.py

```diff
@@ -69,8 +78,29 @@ def validate_property_type(self, property_type: str) -> str:
                 )
             return property_type
 
-    @abstractmethod
-    def retrieve(self, parcl_id: int, params: Optional[Mapping[str, Any]] = None):
+    def validate_portfolio_size(self, portfolio_size: str) -> str:
+        """
+        Validates the portfolio size string and returns it in the expected format.
+        Raises ValueError if the portfolio size is invalid or not in the expected format.
+        """
+        valid_portfolio_sizes = self._get_valid_portfolio_sizes()
+        if portfolio_size:
+            if portfolio_size.upper() not in valid_portfolio_sizes:
+                raise ValueError(
+                    f"Portfolio size {portfolio_size} is not valid. Must be one of {', '.join(valid_portfolio_sizes)}."
+                )
+            return portfolio_size.upper()
+
+    def retrieve(
+        self,
+        parcl_id: int,
+        start_date: str = None,
+        end_date: str = None,
+        property_type: str = None,
+        portfolio_size: str = None,
```

Comment:
Is there any way we can alert to the customization of parameters by endpoint here without having to do a major refactor? for instance, proprety_type and portfolio_size and not accepted params for many endpoints.

Reply:
We also need an auto paginate mechanism defaulted to False. This is needed for endpoints like `pricefeed`